api/classes/base_controller.js

```diff
@@ -11,133 +13,165 @@ function defaultFormatResponse(model) {
   return model;
 }
 
-function BaseController(model)  {
-  this.Model = model;
-}
+class BaseController {
+  constructor(Model) {
+    this.Model = Model;
+  }
 
-BaseController.prototype.formatRequestData = function(req) {
-  // Abstract base method,
-  // formats data for database entry
-};
+  /* eslint no-unused-vars: ["error", { "argsIgnorePattern": "^request$" }]*/
+  formatRequestData(request) {
+    // Abstract base method; formats data for database entry
+  }
 
-BaseController.prototype.formatResponseData = function(model) {
-  // Base method to format data that will be sent in the response
-  // By default, it returns the model as is
-  return model;
-};
-
-BaseController.prototype.getOne = function(req, reply) {
-  var record = this.formatResponseData(req.pre.records.models[0].toJSON());
-  reply(req.generateResponse(record)
-    .code(200));
-};
-
-BaseController.prototype.getAll = function(req, reply) {
-  var records = req.pre.records.toJSON().map(this.formatResponseData);
-  reply(req.generateResponse(records));
-};
-
-// `formatResponse` is an optional processing function that can be passed
-// in to modify what is sent in the response body. If no function is
-// provided, the full model for the current method is used in the response.
-BaseController.prototype.update = function(req, reply, formatResponse) {
-  var reqData = this.formatRequestData(req);
-  formatResponse = typeof formatResponse === 'function' ? formatResponse : defaultFormatResponse;
-
-  var result = Promise.resolve().then(function() {
-    var record = req.pre.records.models[0];
-
-    record.set(reqData);
-    if (!record.hasChanged()) {
-      return record;
-    }
+  formatResponseData(model) {
+    // Base method to format data that will be sent in the response
+    // By default, it returns the model as is
+    return model;
+  }
+
+  // `formatResponse` is an optional processing function that can be passed
+  // in to modify what is sent in the response body. If no function is
+  // provided, the full model for the current method is used in the response.
+  create(request, reply, formatResponse) {
+    const requestData = this.formatRequestData(request);
+
+    formatResponse = typeof formatResponse === `function` ? formatResponse : defaultFormatResponse;
```

Comment:
this seems an odd thing to do: what are we testing here? why not be more explicit here and say:
```
if (typeof formatResponse !== "function) {
  formatResponse = defaultFormatResponse
}
```
rather than a blind reassigment that might reassign to the same reference.